lib/galaxy/tools/__init__.py

```diff
@@ -825,6 +829,35 @@ def tests(self):
             self.__tests_populated = True
         return self.__tests
 
+    def test_data_path(self, filename):
+        repository_dir = None
+
+        # If a tool shed tool, try to find where the repository was installed. Is there a cleaner way
+        # to do this @mvdbeek?
+        if hasattr(self, 'tool_shed') and self.tool_shed:
```

Comment:
@mvdbeek Does this piece of code to find a shed tool's test-data directory from inside of Galaxy make sense to you? In particular is there a better way to recover what I am calling ``repository_dir`` here - the ``test-data`` stuff a little bit lower is Greg code from the test framework so I assume that is right.

Reply:
So this is code stolen from the previous method of resolving tool shed test data - https://github.com/galaxyproject/galaxy/blob/dev/test/base/tool_shed_util.py#L25. I guess there is descended from the repository directory in case the test-data was found next to a tool installed below the repository top of the repository:

```
repo/
- mytools/
  - tool1_1.xml
  - tool1_2.xml
  - test-data/
```

In that case though you are right the test-data would still live next to the tool. My thought was that it would be good to move to the top of the directory in case tools are stored in directories below the top-level but the test-data is stored at the top-level. Repositories like this:

```
repo/
- README.txt
- test-data/
- mytools1/
  - tool1_1.xml
  - tool1_2.xml
- mytools2/
  - tool2_1.xml
```

That seems like a reasonable repository layout for a repository with a lot of tools - I don't know though. What do you think? I'm perfectly willing to just let IUC declare that test-data needs to be right next to the tool in the repository bundle for this to work - that is fine to me.

Reply:
I see, does this alternative layout work with planemo ? If it doesn't I'd assume it's not a very common layout, in which case we could perhaps first check if `filename` exists in self.tool_dir before doing something more complicated. 

wrt `repository_dir` we've got this:
https://github.com/galaxyproject/galaxy/blob/a2b8e445d85b50265da52c06f017c9c78be953b6/lib/galaxy/model/tool_shed_install/__init__.py#L294-L301
I'm not sure the logic of relying on `self.installed_changeset_revision` will work in all cases (Could perhaps explain a subset of help images missing every now and then?!)

Reply:
https://github.com/galaxyproject/planemo/blob/master/planemo/galaxy/config.py#L999

Planemo would allow use of both the ``pwd`` and the tool directory if they are different. So I think I imagined it working this way previously. Also --test-data is configurable on the command-line for testing and can be placed where ever in the suite using .shed.yml. Happy to call it unnecessary complex though.

Reply:
Alright, that makes sense. I don't think it's too complex, this offers some nice flexibility.